pkg/daemon/update.go

```diff
@@ -1158,47 +1158,39 @@ func (dn *CoreOSDaemon) switchKernel(oldConfig, newConfig *mcfgv1.MachineConfig)
 		return nil
 	}
 
-	// Do nothing if both old and new KernelType are of type default
-	if canonicalizeKernelType(oldConfig.Spec.KernelType) == ctrlcommon.KernelTypeDefault && canonicalizeKernelType(newConfig.Spec.KernelType) == ctrlcommon.KernelTypeDefault {
+	oldKtype := canonicalizeKernelType(oldConfig.Spec.KernelType)
+	newKtype := canonicalizeKernelType(newConfig.Spec.KernelType)
+
+	// In the OS update path, we removed overrides for kernel-rt.  So if the target (new) config
+	// is also default (i.e. throughput) then we have nothing to do.
+	if newKtype == ctrlcommon.KernelTypeDefault {
 		return nil
 	}
 
+	// TODO: Drop this code and use https://github.com/coreos/rpm-ostree/issues/2542 instead
 	defaultKernel := []string{"kernel", "kernel-core", "kernel-modules", "kernel-modules-extra"}
```

Comment:
I missed to ask this, since we are not yet adding `kernel-modules-core` in defaultKernel list packages. Fixing OCPBUGS-8113 will still need that, correct?